cassandra/src/main/scala/com/stratio/crossdata/connector/cassandra/CassandraQueryProcessor.scala

```diff
@@ -120,21 +140,25 @@ class CassandraQueryProcessor(cassandraRelation: CassandraXDSourceRelation, logi
   }
 
 
-  def validatedNativePlan: Option[(Array[Attribute], Array[SourceFilter], Map[Attribute, NativeUDF], Option[Int])] = {
+  def validatedNativePlan: Option[CassandraPlan] = {
     lazy val limit: Option[Int] = logicalPlan.collectFirst { case Limit(Literal(num: Int, _), _) => num }
 
-    def findProjectsFilters(lplan: LogicalPlan):
-    (Array[Attribute], Array[SourceFilter], Map[Attribute, NativeUDF], Boolean) = {
+    def findCassandraPlan(lplan: LogicalPlan): (BaseLogicalPlan, Boolean) = {
```

Comment:
Once the refactor has been performed over `validatedNativePlan` return value, why keep the boolean field? I think it may be better to return an `Option`: `def findCassandraPlan(lplan: LogicalPlan): Option[BaseLogicalPlan]`